### PR TITLE
[ISSUE #5994] [RIP-46] add pop and timer metrics

### LINKED ISSUE #5994: [RIP-46] Add pop and timer metrics
This issue aims to add the following metrics:

| Type      | Name                                     | Unit         |
|-----------|------------------------------------------|--------------|
| Histogram | rocketmq_pop_buffer_scan_time_consume    | milliseconds |
| Counter   | rocketmq_pop_revive_in_message_total     |              |
| Counter   | rocketmq_pop_revive_out_message_total    |              |
| Counter   | rocketmq_pop_revive_retry_messages_total |              |
| Gauge     | rocketmq_pop_revive_lag                  |              |
| Gauge     | rocketmq_pop_revive_latency              | milliseconds |
| Gauge     | rocketmq_pop_offset_buffer_size          |              |
| Gauge     | rocketmq_pop_checkpoint_buffer_size      |              |
| Counter   | rocketmq_timer_enqueue_total             |              |
| Counter   | rocketmq_timer_dequeue_total             |              |
| Gauge     | rocketmq_timer_enqueue_lag               |              |
| Gauge     | rocketmq_timer_enqueue_latency           | milliseconds |
| Gauge     | rocketmq_timer_dequeue_lag               |              |
| Gauge     | rocketmq_timer_dequeue_latency           | milliseconds |
| Gauge     | rocketmq_timing_messages                 |              |

### PR DESCRIPTION
**Pull request by ShadowySpirits**
close #5994

### CODE REVIEW

File: broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java [hunk 1]
```diff
@@ -66,6 +67,7 @@ public class PopReviveService extends ServiceThread {
     private int queueId;
     private BrokerController brokerController;
     private String reviveTopic;
+    private volatile long currentReviveMessageTimestamp = -1;
```
Comment (lizhimins) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java:70]: Need to add the volatile modifier here?

File: broker/src/test/java/org/apache/rocketmq/broker/processor/PopReviveServiceTest.java [hunk 2]
```diff
@@ -249,22 +249,24 @@ public static MessageExtBrokerInner buildCkMsg(PopCheckPoint ck) {
...
-    public static MessageExtBrokerInner buildAckMsg(AckMsg ackMsg, long deliverMs, long reviveOffset, long deliverTime) {
+    public static MessageExtBrokerInner buildAckMsg(AckMsg ackMsg, long deliverMs, long reviveOffset,
+        long deliverTime) {
         MessageExtBrokerInner messageExtBrokerInner = buildAckInnerMessage(
-                REVIVE_TOPIC,
-                ackMsg,
-                REVIVE_QUEUE_ID,
-                STORE_HOST,
-                deliverMs,
-                PopMessageProcessor.genAckUniqueId(ackMsg)
```
Comment (lizhimins) [inline broker/src/test/java/org/apache/rocketmq/broker/processor/PopReviveServiceTest.java:259]: code style is strange..

File: store/src/main/java/org/apache/rocketmq/store/timer/TimerMessageStore.java [hunk 3]
```diff
@@ -1106,6 +1106,13 @@ private MessageExtBrokerInner convertMessage(MessageExt msgExt, boolean needRoll
         return msgInner;
     }
 
+    private String getRealTopic(MessageExt msgExt) {
+        if (msgExt == null) {
+            return null;
+        }
+        return msgExt.getProperty(MessageConst.PROPERTY_REAL_TOPIC);
```
Comment (lizhimins) [inline store/src/main/java/org/apache/rocketmq/store/timer/TimerMessageStore.java:1113]: Here may return null

**Review by xdkxlk — APPROVED** [review]
LGTM

**Pull request merged**